pages/insight_menu.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+function echoCategory($categoryId, $size = 1) {
+    $subcategories = base_query("SELECT * FROM DishCategory WHERE ParentCategoryId = :categoryId ORDER BY Position", [':categoryId' => $categoryId])->fetchAll();
+    $category = base_query("SELECT * FROM DishCategory WHERE Id = :categoryId", [':categoryId' => $categoryId])->fetch();
+    ?>
+
+<!-- Echo category name -->
+<div style="margin:<?= $size * 10 ?>px">
+    <h<?= $size ?>>
+        <?= $category['Name']?>
+        <!-- Checks if a category has a price attached to itself -->
+        <?php if(isset($category['Price'])) {
+            ?><span id='categoryPrice'><?= $category['Price'] ?></span><?php
+        } ?>
+    </h<?= $size ?>> 
+
+<!-- Echo category description -->
+     <i><p>
+        <?= $category['Description'] ?>
+    </p></i> 
+
+<?php
+
+    // If there are still subcategories the function will keep being called upon
+    $dishes = base_query("SELECT * FROM Dish WHERE Category = :categoryId ORDER BY Position", [':categoryId' => $categoryId])->fetchAll();
+    // var_dump($dishes);
+    foreach($dishes as $dishValue){ ?>
+       <ul> <?php
```

Comment:
Je opent en sluit direct een PHP tag. Die kun je weg halen. Op de volgende lijn is dit ook het geval.